packages/di/README.md

```diff
@@ -216,3 +216,27 @@ If at some point you want to create registry that wan't be overrided just call t
 ```ts
 const boldRegistry = new Registry({ id: cnApp(), overridable: false })
 ```
+
+## Extends components
+
+Components inside registry can be extends (e.g. for experiments) by wrapping `withRegistry(...)(App)` with another registry.
+
+```tsx
+import { Registry, withRegistry, withBase } from '@bem-react/di'
+
+import { AppDesktop, registryId } from './App@desktop'
+
+const expRegistry = new Registry({ id: registryId })
+
+// extends original Header
+expRegistry.set('Header', withBase(BaseHeader => props => (
+  <div>
+    <Header height={200} color={red}/>
+  </div>
+)))
+
+// AppDesktopExperimental will call App with extended 'Header'
+export const AppDesktopExperimental = withRegistry(expRegistry)(AppDesktop)
+```
+
+When `App` extracts components from registry _DI_ actually takes all registries defined above and merges. By default higher defined registry extends lower defined one.
```

Comment:
_DI_ merges nested registries composing `withBase`-forms and ordinary components for you.

Reply:
...So you always can get a reference to previous component's implementation.